torchvision/datasets/samplers/clip_sampler.py

```diff
@@ -164,11 +172,10 @@ def __iter__(self):
             sampled = torch.randperm(length)[:size] + s
             s += length
             idxs.append(sampled)
-        idxs = torch.cat(idxs)
```

Comment:
This scheme of overriding the variable type was quite common here. In this example `idxs` was `List[torch.Tensor]` before, `torch.Tensor` here, and `List[int]` later. Thus, I need to rename some variables or merge statements to avoid assignments all together. Otherwise, this will be an even more complicated mess with a lot of `cast`ing.